qiskit/dagcircuit/dagdepnode.py

```diff
@@ -161,12 +121,8 @@ def semantic_eq(node1, node2):
                 if node1.name == node2.name:
                     if node1._qargs == node2._qargs:
                         if node1.cargs == node2.cargs:
-                            if node1.type == "op":
-                                if getattr(node1._op, "condition", None) != getattr(
-                                    node2._op, "condition", None
-                                ):
-                                    return False
-                            return True
+                            if node1._op.condition == node2._op.condition:
+                                return True
```

Comment:
This change isn't related, also looks wrong - we still need to check they're `op` nodes, and the `getattr` stuff is because the `Operation` interface doesn't define `condition`, so stuff isn't guaranteed to have it.  Strictly, it also affects the order of early return to be incorrect, which at the moment would have no effect, but could lead to bugs if more conditions were added later.  I'd revert this.

```suggestion
                            if node1.type == "op":
                                if getattr(node1._op, "condition", None) != getattr(
                                    node2._op, "condition", None
                                ):
                                    return False
                            return True
```